bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -510,6 +501,17 @@ private WidgetDTO createWidgetBean(String sitemapName, Widget widget, boolean dr
         return bean;
     }
 
+    private String buildProxyUrl(String sitemapName, Widget widget, URI uri) {
+        String wId = itemUIRegistry.getWidgetId(widget);
+        StringBuilder sb = new StringBuilder();
+        sb.append(uri.getScheme()).append("://").append(uri.getHost());
+        if (uri.getPort() >= 0 && uri.getPort() != 80) {
```

Comment:
You moved this method only to another method, but is it really correct?
If the scheme is http the default port is 80, so all is fine.
If the scheme is https the default port should be 443 and so `uri.getPort()` should be checked to be 443.
Okay, this only affect the strange situation if someone runs https on port 80, so we should perhaps never run into.
But shouldn't it be fixed?

Reply:
We could set the port in URL everytime ?

Reply:
I agree. If it is the default one, it does not matter..